src/bitfield/mod.rs

```diff
@@ -23,39 +23,102 @@ mod masks;
 use self::masks::Masks;
 use flat_tree::{self, Iterator as FlatIterator};
 pub use sparse_bitfield::{Bitfield as SparseBitfield, Change};
+use std::convert::TryInto;
 use std::ops::Range;
 
 /// Bitfield with `{data, tree, index} fields.`
 #[derive(Debug)]
 pub struct Bitfield {
     data: SparseBitfield,
-    /// FIXME: SLEEP protocol tree field.
-    pub tree: SparseBitfield,
     index: SparseBitfield,
     page_len: u64,
     length: u64,
     masks: Masks,
     iterator: FlatIterator,
 }
 
-impl Default for Bitfield {
-    fn default() -> Self {
-        Bitfield::new()
-    }
-}
-
 impl Bitfield {
     /// Create a new instance.
-    pub fn new() -> Self {
-        Self {
+    pub fn new() -> (Self, SparseBitfield) {
```

Comment:
I'm not sure if I get why we are returning two structs instead of keeping it inside the Bitfield structure.
If they are associated I feel like they should remain on a single structure. Would you mind sharing more about this change, please?

Reply:
its because i pass returned `SparseBitfield` into TreeIndex here:
https://github.com/datrs/hypercore/pull/121/files#diff-75125abd63ba614d4c111cb5e5e27dd5R49-R56

Reply:
this is also the reason why I pass tree into `to_bytes()` here
https://github.com/datrs/hypercore/pull/121/files#diff-409aa10a28c67440a4f460f8e4800372R169

Reply:
Got it. Maybe the feed could avoid having the `tree` index and use the `self.bitfield.tree` directly to avoid needing to have a `new` function that returns a tuple. What do you think?

That could be a larger refactor later tho.